### PR TITLE
fix(source-shopify): Add `checkpointed value` collision detection for BULK-checkpoint-supported streams

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving:
- https://github.com/airbytehq/oncall/issues/6874

## How
- New Exception: Added BulkJobCheckpointCollisionError in bulk/exceptions.py.
- Checkpoint Collision Detection: Added methods to handle and check for checkpoint collisions in job.py. This includes setting the last checkpoint cursor value and checking for collisions during the job process.
- Spec update: Modified spec.json to update the description and range for job_checkpoint_interval to 15k-1M rows.
Unit Test Update:
- Edited the test case in `test_bulk_stream.py` to handle the scenario of repeated slices and checkpoint collisions.

These changes collectively improve the handling of checkpoint collisions in BULK-supported streams, ensuring more robust and error-free data synchronization.

## User Impact
No impact is expected. This is not a breaking change.

## Can this PR be safely reverted and rolled back?
- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by bazarnov** [comment]
The `regression tests` failed due to the `floating` nature of the following object:

The `products stream`:

```json
{
  "values_changed": {
    "root[252]['data']['onlineStorePreviewUrl']": {
      "new_value": "string_len-372_35b7a6dd1f37b5d995986a0ce5a0d65a4d6009e81b8dbda9a33954e06937d39a",
      "old_value": "string_len-372_0bcb2b6c419fa816637b9e331f307f990255855c7bee47759cd0ea3a148e8971"
    }
  }
}
```

This considered as `safe` to go with.

File: airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py [hunk 1]
```diff
@@ -498,8 +521,21 @@ def get_adjusted_job_start(self, slice_start: datetime) -> datetime:
...
         """
-        return pdm.parse(checkpointed_cursor) if checkpointed_cursor else slice_end
+
+        if checkpointed_cursor:
+            self._set_last_checkpoint_cursor_value(checkpointed_cursor)
+
+            if self._checkpoint_cursor_has_collision(checkpointed_cursor):
+                raise ShopifyBulkExceptions.BulkJobCheckpointCollisionError(
+                    f"The stream: `{self.http_client.name}` checkpoint collision is detected. Try to increase the `BULK Job checkpoint (rows collected)` to the bigger value. The stream will be synced again during the next sync attempt."
+                )
```
Comment (aldogonzalez8) [inline airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py:534]: Is not this setting `checkpointed_cursor` to `self._job_last_checkpoint_cursor_value` and later checking If same `checkpointed_cursor` value is equal to `self._job_last_checkpoint_cursor_value`?

Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py:534]: Thanks. Corrected!

**Review by aldogonzalez8 — APPROVED** [review]
APPROVED

**Pull request merged**